xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -653,9 +659,21 @@ where
         intent: &StoredGroupIntent,
         message: &ProtocolMessage,
         envelope: &GroupMessageV1,
-    ) -> Result<(IntentState, Option<Vec<u8>>), GroupMessageProcessingError> {
-        if intent.state == IntentState::Committed {
-            return Ok((IntentState::Committed, None));
+    ) -> Result<Option<Vec<u8>>, IntentResolutionError> {
+        if intent.state == IntentState::Committed
+            || intent.state == IntentState::Processed
+            || intent.state == IntentState::Error
+        {
+            tracing::warn!(
+                "Skipping already processed intent {} of kind {} because it is in state {:?}",
+                intent.id,
+                intent.kind,
+                intent.state
+            );
+            return Err(IntentResolutionError {
+                processing_error: GroupMessageProcessingError::IntentAlreadyProcessed,
+                next_intent_state: intent.state,
+            });
```

Comment:
I believe this is an alternative way to address https://github.com/xmtp/libxmtp/pull/1899, but would be good for @neekolas to double-check. If the intent is already Committed or Processed, we return the same state as a non-retriable error, and skip the rest of the function. The outer function will just skip over the payload seeing as the state didn't change. This allowed me to simplify the return types and expected behavior - if this function returns `Ok()`, we are always transitioning the intent to `Committed`.

Reply:
This seems like a nice cleanup as well